distributed/protocol/compression.py

```diff
@@ -39,56 +41,35 @@
 with suppress(ImportError):
     import snappy
 
-    def _fixed_snappy_decompress(data):
-        # snappy.decompress() doesn't accept memoryviews
-        if isinstance(data, (memoryview, bytearray)):
-            data = bytes(data)
-        return snappy.decompress(data)
+    # In python-snappy 0.5.3, support for the Python Buffer Protocol was added.
+    # This is needed to handle other objects (like `memoryview`s) without
+    # copying to `bytes` first.
+    #
+    # Note: `snappy.__version__` doesn't exist in a release yet.
+    #       So do a little test that will fail if snappy is not 0.5.3 or later.
+    try:
+        snappy.compress(memoryview(b""))
+    except TypeError:
+        raise ImportError("Need snappy >= 0.5.3")
```

Comment:
Recently `snappy.__version__` was added with PR ( https://github.com/andrix/python-snappy/pull/119 ). Though it is not currently in a release and hasn't been around long enough to rely upon.

So just do a simple test that will fail without `python-snappy` version `0.5.3` or later. If `python-snappy` is not new enough (unlikely given how long this release has been out), simply skip using `python-snappy`.

Reply:
+1

(do we want to use cramjam for all of the de/compressors??)